Sources/Sentry/SentryProfiler.mm

```diff
@@ -373,6 +387,25 @@ + (void)stopProfilerForReason:(SentryProfilerTruncationReason)reason
     _gCurrentProfiler = nil;
 }
 
+- (void)startMetricProfiler
+{
+    if (_gCurrentSystemWrapper == nil) {
+        _gCurrentSystemWrapper = [[SentrySystemWrapper alloc] init];
+    }
+    if (_gCurrentProcessInfoWrapper == nil) {
+        _gCurrentProcessInfoWrapper = [[SentryNSProcessInfoWrapper alloc] init];
+    }
+    if (_gCurrentTimerWrapper == nil) {
+        _gCurrentTimerWrapper = [[SentryNSTimerWrapper alloc] init];
+    }
```

Comment:
`m`: I think we could move that to `init` cause this code is going to be executed shortly after init, I guess.

Reply:
Agreed. I think maybe I initially thought these could be reset from test subclasses by setting them to nil, but that's currently not necessary.

Reply:
Ah. I actually do need these. Otherwise when SentryProfiler initializes a new SentryMetricProfiler, it will override the test subclass instances with these normal instances.